src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -0,0 +1,251 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+# The code in this file is largely a copy of https://github.com/Snowflake-Labs/snowcli/blob/main/src/snowcli/utils.py
+
+import glob
+import os
+import platform
+import subprocess
+import sys
+import zipfile
+from logging import getLogger
+from pathlib import Path
+from typing import Dict, List, Optional, Set, Tuple
+
+import pkg_resources
+from pkg_resources import Requirement
+
+_logger = getLogger(__name__)
+PIP_ENVIRONMENT_VARIABLE = "PIP_NAME"
+IMPLICIT_ZIP_FILE_NAME = "zipped_packages"
+SNOWPARK_PACKAGE_NAME = "snowflake-snowpark-python"
+
+# Default that we do not want to upload because we wish to use stable versions instead
+# TODO: This is very hacky, how can errors with default packages be avoided?
+DEFAULT_PACKAGES = {
```

Comment:
How are default packages selected? It looks like some packages should already be handled by native package detection, so we could probably drop them from here :)

Reply:
True, this is no longer needed because of improved native packager detection